### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by matteofox**
muse_redux_gui.py changed

### CODE REVIEW

File: ifu/muse_redux_gui.py [hunk 1]
```diff
@@ -49,19 +47,37 @@ def reduxgui(listimg,mode='align'):
...
+	self.boxry   = []
+    
+    
     class align_tk(Tkinter.Tk):
      
-        def __init__(self,filename,parent,mode='align',refra=[],refdec=[],refid=[]):
-            self.root=Tkinter.Tk.__init__(self,parent)
+        def __init__(self,filename,parent,guivalues,mode='align',refra=[],refdec=[],refid=[]):
+            
+	    self.root=Tkinter.Tk.__init__(self,parent)
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: Is indenting correct here?

File: ifu/muse_redux_gui.py [hunk 2]
```diff
@@ -150,6 +161,7 @@ def init_dataframe(self):
            
             #now open image
             self.fits=fits.open(self.filename)
+	    self.mjd=self.fits[0].header['MJD-OBS']
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: Same here..

File: ifu/muse_redux_gui.py [hunk 3]
```diff
@@ -230,7 +234,7 @@ def setscale(self,*args):
             self.vmax=float(self.vmaxv.get())
 
             #trigger update
-            self.update_twodimage(update=True)
+	    self.update_twodimage(update=True)
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: and here

File: ifu/muse_redux_gui.py [hunk 4]
```diff
@@ -247,26 +251,27 @@ def update_twodimage(self,update=False):
...
+                self.twodimagePlot_prop["axis"].plot([self.boxrx[ii],self.boxrx[ii]],[self.boxly[ii],self.boxry[ii]],color='red')
+                self.twodimagePlot_prop["axis"].plot([self.boxlx[ii],self.boxrx[ii]],[self.boxry[ii],self.boxry[ii]],color='red')
+                self.twodimagePlot_prop["axis"].plot([self.boxlx[ii],self.boxlx[ii]],[self.boxly[ii],self.boxry[ii]],color='red')
+                self.twodimagePlot_prop["axis"].text(self.boxrx[ii],self.boxry[ii],"{}".format(ii+1),color='red')
+            
+	    if(len(self.refid) > 0):
+                for ii in range(len(self.refid)):
                     rx,ry=self.wcs.wcs_world2pix(self.refra[ii],self.refdec[ii],1)
-                    self.twodimagePlot_prop["axis"].text(rx,ry,"{}".format(ll),color='red')
+                    if rx>0 and ry>0 and rx<=np.shape(self.fitimg)[1] and ry<=np.shape(self.fitimg)[0]:
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: Please keep () around if - makes it easier to read

File: ifu/muse_redux_gui.py [hunk 5]
```diff
@@ -275,12 +280,15 @@ def movemouse(self,event):
             
             """ Do stuff when  mouse moves """
             
-            try:
+	    try:
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: check indenting of this block

File: ifu/muse_redux_gui.py [hunk 6]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
+	      return self.nstars
+	     
+	        
         def presskey(self,event):
             
             """ Do stuff when press  key  """
             
             #quit on q
-            if(event.key == "q"):
+            if(event.key == "q")muse_redux_gui.py:
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: Are you sure about this line or is it a typo?

File: ifu/muse_redux_gui.py [hunk 7]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
-                global boxry
-                global boxlx
-                global boxly
-                boxlx=boxlx[0:-1]
-                boxly=boxly[0:-1]
-                boxrx=boxrx[0:-1]
-                boxry=boxry[0:-1]
-                self.nbox=self.nbox-1
+	    
+	    elif(event.key == "d"):
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: check indenting of this block

File: ifu/muse_redux_gui.py [hunk 8]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
+	    self.guivalues.starra  = np.array(self.starra)[ind]  
+	    self.guivalues.stardec = np.array(self.stardec)[ind]
+	    self.guivalues.starid  = np.array(self.starid)[ind] 
+	    self.guivalues.boxlx = self.boxlx
+	    self.guivalues.boxly = self.boxly
+	    self.guivalues.boxrx = self.boxrx
+	    self.guivalues.boxry = self.boxry
+	    self.guivalues.mjd   = self.mjd
+	    
+	    self.quit()
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: here as well

File: ifu/muse_redux_gui.py [hunk 9]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
+	  	  ra_ref.append(float(temp[0])) 
+          	  dec_ref.append(float(temp[1])) 
+	  	  id_ref.append(ii)
+          
+	  ra_ref  = np.array(ra_ref)
+	  dec_ref = np.array(dec_ref)
+	  id_ref  = np.array(id_ref)
+	   
+	else:
+	  print('REDUXGUI: Run in align mode with first exp as reference')
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: and here...

File: ifu/muse_redux_gui.py [hunk 10]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
                 app.mainloop()
                 ra_off.append(0.0)
                 dec_off.append(0.0)
-                ra_ref=np.copy(starra)
-                dec_ref=np.copy(stardec)
-                id_ref=np.copy(starid)
+                ra_ref=np.copy(GUIvalues.starra)
+                dec_ref=np.copy(GUIvalues.stardec)
+                id_ref=np.copy(GUIvalues.starid)
+		mjd.append(GUIvalues.mjd)
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: and here..

File: ifu/muse_redux_gui.py [hunk 11]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
             else:
                 ##main loop for other exposures
-                app = align_tk(name,None,mode=mode,refra=ra_ref,refdec=dec_ref,refid=id_ref)
+		app = align_tk(name,None,GUIvalues,mode=mode,refra=ra_ref,refdec=dec_ref,refid=id_ref)
                 app.title('Align {}'.format(name))
                 app.mainloop()
-                ra_off.append(np.mean(-ra_ref+starra))
-                dec_off.append(np.mean(-dec_ref+stardec))
-                
+		thisraoff = np.mean(-ra_ref[np.in1d(id_ref,GUIvalues.starid)]+GUIvalues.starra)
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: and here...

File: ifu/muse_redux_gui.py [hunk 12]
```diff
@@ -299,147 +307,197 @@ def movemouse(self,event):
...
         strcut=fits.open('OFFSET_LIST.fits',mode='update')
-       
+        
+	mjdfile = (strcut[1].data[:]['MJD_OBS'])
+	 
         for xx in range(len(dec_off)):
-            strcut[1].data[xx]['RA_OFFSET']=ra_off[xx]
-            strcut[1].data[xx]['DEC_OFFSET']=dec_off[xx]
+            offlistind = np.argmin(abs(mjdfile-mjd[xx]))
+	    strcut[1].data[offlistind]['RA_OFFSET']=ra_off[xx]
```
Comment (mifumagalli) [inline ifu/muse_redux_gui.py]: and here...

**Comment by matteofox** [comment]
Should be ok now. Indentation fixed by replacing tabs with spaces. Typos fixed. Fixed () for if statements.

**Comment by matteofox** [comment]
muse_redux_gui.py still requires a couple of bug fixes:
1) If you change a value in ZRANGE the focus does not return to the main (draw) widget. If you have ideas....
2) Try a smaller centroid range if 20 pixels is too big and your objct is close to the edges of the FoV

**Comment by mifumagalli** [comment]
Closed

**Pull request merged**